src/api.js

```diff
@@ -166,7 +179,12 @@ const Api = (options = {}) => {
         rb,
         hdl: [
           () => parse(request, params, event),
-          paramsOut => handler(paramsOut, context, rb, event)
+          async paramsOut => [
+            await handler(paramsOut, context, rb, event),
+            pruneResponse === true && rawFieldsParam !== undefined && paramsOut[rawFieldsParam.name] !== undefined
+              ? objectPaths.split(paramsOut[rawFieldsParam.name])
+              : null
```

Comment:
This should only prune if it's a json response. Otherwise probably an assertion error?

So these should be chained instead.

I'd further not change the format but rather associate to the response using Symbols like for example [here](https://github.com/blackflux/object-scan/blob/master/src/util/compiler.js#L9)